Add force parameter (allows to scroll only if element is not in view)

v1r0x opened the pull request:
Refs #55 

I added a new parameter (`force`) to the options. It defaults to `true` and the behavior should be the same. If set to `false`, it only scrolls if the element is not in view (same as `scrollIntoView()` referenced in #55).

This should fix the first bullet in #55
IMO the second bullet can be done using the `offset` param.

Hope it's useful :)

cc @rigor789

rigor789 commented:
Thanks for the PR! 🎉 

Does this work in nested/different containers?

[There is a `cumulativeOffset` utility](https://github.com/rigor789/vue-scrollto/blob/master/src/utils.js#L39-L53) that correctly calculates the offset when there are nested scrollable elements, and I believe that it should be used here.

Docs would also need updating to document the property and it's behavior.

v1r0x commented:
I totally forgot to update the docs 😊 Wanted to do that before I open the PR.

I tested it with `<li>` inside `<ul>` inside `<div>` inside `<div id="container">` and it works

rigor789 commented:
Can you test your changes against a sample like this? https://jsfiddle.net/ns0dkqte/

I believe it will act weird if you don't use the `cumulativeOffset` helper.

v1r0x commented:
The example works fine. The `cumulativeOffset` calculates a different offset from what I use.

- `containerTop`: Is the position/offset of the scroll bar (0 if not scrolled)
- `containerBottom`: Is the height of the container (`50vh` in px in your example) + the offset `containerTop`
- `elementTop`: Is `The HTMLElement.offsetTop read-only property returns the distance of the current element relative to the top of the offsetParent node.` (from [offsetTop](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop)). There might be some other (recursive) approach to get the offset inside the container, but it seems to work with this method. If you have some other approach, I'm happy to try it
- `elementBottom`: Is `elementTop` + the height of the element, so it scrolls, if the element is only in parts in view.

v1r0x commented:
Understandable :+1: 
Playing with offsets is the same as timezones. It's always sh*t :D

The example from #11 works fine, too.
I added the parameter to the readme and also updated the code. Some of my calculations are already done by your existing code. Thus, I moved and adjusted my code to use them instead of re-calculate them.
